cumulus/parachains/integration-tests/emulated/tests/assets/asset-hub-westend/src/tests/set_asset_claimer.rs

```diff
@@ -0,0 +1,217 @@
+		assert_ok!(RuntimeCall::PolkadotXcm(pallet_xcm::Call::execute {
+			message: bx!(VersionedXcm::from(local_xcm)),
+			max_weight: Weight::from_parts(4_000_000_000_000, 300_000),
+		})
+		.dispatch(test.signed_origin));
+	});
+
+	let balance_after_trap =
+		<AssetHubWestend as Chain>::account_data_of(alice_account.clone()).free;
+	assert_eq!(alice_balance_after - balance_after_trap, amount_to_send);
+
+	let bob_balance_before = <AssetHubWestend as Chain>::account_data_of(bob_account.clone()).free;
+	let test_args = TestContext {
+		sender: bob_account.clone(),
+		receiver: alice_account.clone(),
+		args: TestArgs::new_para(
+			alice_location.clone(),
+			alice_account.clone(),
+			amount_to_send,
+			assets.clone(),
+			None,
+			0,
+		),
+	};
+	let test = SystemParaToParaTest::new(test_args);
+	let xcm = Xcm::<RuntimeCall>::builder_unsafe()
+		.claim_asset(test.args.assets.clone(), Here)
+		.deposit_asset(AllCounted(test.args.assets.len() as u32), bob_location.clone())
+		.build();
+
+	AssetHubWestend::execute_with(|| {
+		assert_ok!(RuntimeCall::PolkadotXcm(pallet_xcm::Call::execute {
+			message: bx!(VersionedXcm::from(xcm)),
+			max_weight: Weight::from_parts(4_000_000_000_000, 300_000),
+		})
+		.dispatch(test.signed_origin));
+	});
+
+	let bob_balance_after = <AssetHubWestend as Chain>::account_data_of(bob_account.clone()).free;
+	assert_eq!(bob_balance_after - bob_balance_before, amount_to_send);
+}
+
+fn account_and_location(account: &str) -> (AccountId32, Location) {
+	let account_id = AssetHubWestend::account_id_of(account);
+	let account_clone = account_id.clone();
+	let location: Location =
+		[Junction::AccountId32 { network: Some(Westend), id: account_id.into() }].into();
+	(account_clone, location)
+}
+
+// The test:
+// 1. Funds Bob account on BridgeHub, withdraws the funds, sets asset claimer to
+// sibling account of Alice on AssetHub and traps the funds.
+// 2. Sends an XCM from AssetHub to BridgeHub on behalf of Alice. The XCM: claims assets,
+// pays fees and deposits assets to alice's sibling account.
+#[test]
+fn test_sac_between_the_chains() {
+	let alice = AssetHubWestend::account_id_of(ALICE);
+	let alice_bh_sibling = Location::new(
+		1,
+		[
+			Parachain(AssetHubWestend::para_id().into()),
+			Junction::AccountId32 { network: Some(Westend), id: alice.clone().into() },
+		],
+	);
+	let bob = BridgeHubWestend::account_id_of(BOB);
+	let bob_location =
+		Location::new(0, Junction::AccountId32 { network: Some(Westend), id: bob.clone().into() });
+
+	let amount_to_send = 16_000_000_000_000u128;
+	BridgeHubWestend::fund_accounts(vec![(bob.clone(), amount_to_send * 2)]);
+
+	let assets: Assets = (Parent, amount_to_send).into();
+	let test_args = TestContext {
+		sender: bob.clone(),
+		receiver: alice.clone(),
+		args: TestArgs::new_para(
+			bob_location.clone(),
+			bob.clone(),
+			amount_to_send,
+			assets.clone(),
+			None,
+			0,
+		),
+	};
+	let test = BridgeHubToAssetHubTest::new(test_args);
```

Comment:
same as before, you don't need this `TestContext/BridgeHubToAssetHubTest`, I think it just makes the test harder to read and understand.

Reply:
removed